scene-lib/src/scenelib/annotations/el/LocalLocation.java

```diff
@@ -1,91 +1,172 @@
 package scenelib.annotations.el;
 
+import org.objectweb.asm.Label;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
 import java.util.Objects;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
 /**
- * A {@link LocalLocation} holds location information for a local
- * variable: slot index, scope start, and scope length.
+ * A {@link LocalLocation} holds information about a local variable.
+ * A variable may have multiple lifetimes.
+ * We store this information the same way ASM does, as 3 parallel arrays.
  */
 public final class LocalLocation {
     /**
-     * The slot index of the local variable.
+     * The starts of the lifetimes of the element being visited.
```

Comment:
The documentation of all 4 fields contains "being visited", but `LocalLocation` has no notion of visiting.